kotlin-asyncapi-annotation/src/main/kotlin/org/openfolder/kotlinasyncapi/annotation/channel/ChannelBinding.kt

```diff
@@ -0,0 +1,119 @@
+    val mqtt: ChannelBinding.MQTT = ChannelBinding.MQTT(default = true),
+    val mqtt5: ChannelBinding.MQTT5 = ChannelBinding.MQTT5(default = true),
+    val nats: ChannelBinding.NATS = ChannelBinding.NATS(default = true),
+    val jms: ChannelBinding.JMS = ChannelBinding.JMS(default = true),
+    val sns: ChannelBinding.SNS = ChannelBinding.SNS(default = true),
+    val solace: ChannelBinding.Solace = ChannelBinding.Solace(default = true),
+    val sqs: ChannelBinding.SQS = ChannelBinding.SQS(default = true),
+    val stomp: ChannelBinding.STOMP = ChannelBinding.STOMP(default = true),
+    val redis: ChannelBinding.Redis = ChannelBinding.Redis(default = true),
+    val mercure: ChannelBinding.Mercure = ChannelBinding.Mercure(default = true),
+    val ibmmq: ChannelBinding.IBMMQ = ChannelBinding.IBMMQ(default = true)
+)
+
+sealed interface ChannelBinding {
+
+    annotation class AnypointMQ(
+        val default: Boolean = false,
+        val destination: String = "",
+        val destinationType: String = "",
+        val bindingVersion: String = ""
+    )
+
+    annotation class AMQP(
+        val default: Boolean = false,
+        val `is`: String = "",
+        val bindingVersion: String = "",
+        val exchange: AMQPExchange = AMQPExchange(default = true),
+        val queue: AMQPQueue = AMQPQueue(default = true)
+    )
+
+    annotation class AMQPExchange(
+        val default: Boolean = false,
+        val name: String = "",
+        val type: String = "",
+        val durable: Boolean = false,
+        val autoDelete: Boolean = false,
+        val vhost: String = ""
+    )
+
+    annotation class AMQPQueue(
+        val default: Boolean = false,
+        val name: String = "",
+        val type: String = "",
+        val durable: Boolean = false,
+        val exclusive: Boolean = false,
+        val autoDelete: Boolean = false,
+        val vhost: String = ""
+    )
+
+    annotation class IBMMQ(
+        val default: Boolean = false,
+        val destinationType: String = "",
+        val bindingVersion: String = "",
+        val maxMsgLength: Int = 0,
+        val queue: IBMMQQueue = IBMMQQueue(default = true, objectName = ""),
+        val topic: IBMMQTopic = IBMMQTopic(default = true)
+    )
+
+    annotation class IBMMQQueue(
+        val default: Boolean = false,
+        val objectName: String,
+        val isPartitioned: Boolean = false,
+        val exclusive: Boolean = false
+    )
+
+    annotation class IBMMQTopic(
+        val default: Boolean = false,
+        val string: String = "",
+        val objectName: String = "",
+        val durablePermitted: Boolean = false,
+        val lastMsgRetained: Boolean = false
+    )
+
+    annotation class WebSockets(
+        val default: Boolean = false,
+        val method: String = "",
+        val bindingVersion: String = "",
+        val query: Schema = Schema(default = true),
+        val headers: Schema = Schema(default = true)
+    )
+
```

Comment:
Why is sqs/sns missing here?

Reply:
Because AsyncAPI does not define Channel bindings for SQS/SNS https://github.com/asyncapi/bindings/tree/master/sqs